File: examples/src/keyboard-events/c/keyboard_events.c
#define USING_NAMESPACE_P8G
#include "p8g.h"

#include <stdio.h>
#include <string.h>

char buf[16] = "Press any key";

void draw() {
    background(0);
    fill(255);
    textSize(32);
    text(buf, width / 2.f - (strlen(buf) * 18.f) / 2.f, height / 2.f + 8.f);
}

void keyPressed() {
    snprintf(buf, 16, "keyCode: %d", keyCode);
}

int main() {
    run(320, 320, "Keyboard events");
}

File: examples/src/keyboard-events/js/keyboard-events.js
import p8g, {
  background,
  createCanvas,
  fill,
  height,
  keyCode,
  text,
  textSize,
  width,
} from 'p8g.js';

let str = 'Press any key';

p8g.draw = () => {
  background(0);
  fill(255);
  textSize(32);
  text(str, width / 2 - (str.length * 18) / 2, height / 2 + 8);
};

p8g.keyPressed = () => {
  str = `keyCode: ${keyCode}`;
};

createCanvas(320, 320);
